data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java

```diff
@@ -21,52 +22,92 @@
     private Jedis jedisMock = new MockJedis("test");
     private Jedis deviceJedisMock = new MockJedis("device");
+    private Connection connectionMock;
+    private Statement statementMock;
+    private Statement statement;
     private DeviceProfileUpdaterService deviceProfileUpdaterService;
     private DeviceProfileUpdaterSink deviceProfileUpdaterSinkMock;
     private Config configMock;
     private Integer deviceStoreId = 2;
+    private String postgres_table;
     private Gson gson = new Gson();
     private Type mapType = new TypeToken<Map<String, Object>>() { }.getType();
 
     @Before
-    public void setUp() {
+    public void setUp() throws Exception {
         redisConnectMock = mock(RedisConnect.class);
-        cassandraConnectMock=mock(CassandraConnect.class);
+        postgresConnectMock=mock(PostgresConnect.class);
 
         deviceProfileUpdaterSinkMock = mock(DeviceProfileUpdaterSink.class);
         configMock = mock(Config.class);
+        connectionMock = mock(Connection.class);
+        statementMock = mock(Statement.class);
         stub(redisConnectMock.getConnection(deviceStoreId)).toReturn(deviceJedisMock);
+        stub(postgresConnectMock.getConnection()).toReturn(connectionMock);
+        stub(postgresConnectMock.resetConnection()).toReturn(connectionMock);
         envelopeMock = mock(IncomingMessageEnvelope.class);
         stub(configMock.getInt("redis.database.deviceStore.id", deviceStoreId)).toReturn(deviceStoreId);
         stub(configMock.get("input.device.topic.name","device.profile")).toReturn("events_deviceprofile");
+        postgres_table = "device_profile";
+
+        EmbeddedPostgres pg = EmbeddedPostgres.start();
+        Connection con = pg.getPostgresDatabase().getConnection();
+        statement = con.createStatement();
+
+        stub(connectionMock.createStatement()).toReturn(statement);
+        statement.execute("CREATE TABLE device_profile(\n" +
+                "   device_id text PRIMARY KEY,\n" +
+                "   api_last_updated_on TIMESTAMP,\n" +
+                "    avg_ts float,\n" +
+                "    city TEXT,\n" +
+                "    country TEXT,\n" +
+                "    country_code TEXT,\n" +
+                "    device_spec json,\n" +
+                "    district_custom TEXT,\n" +
+                "    fcm_token TEXT,\n" +
+                "    first_access TIMESTAMP,\n" +
+                "    last_access TIMESTAMP,\n" +
+                "    producer_id TEXT,\n" +
+                "    state TEXT,\n" +
+                "    state_code TEXT,\n" +
+                "    state_code_custom TEXT,\n" +
+                "    state_custom TEXT,\n" +
+                "    total_launches bigint,\n" +
+                "    total_ts float,\n" +
+                "    uaspec json,\n" +
+                "    updated_date TIMESTAMP,\n" +
+                "    user_declared_district TEXT,\n" +
+                "    user_declared_state TEXT)");
```

Comment:
Test cases are missing for user_declared_on field

Reply:
Added test cases.